### PR TITLE
feat(setupcheck): Add setup check for maintenance_window_start config

### PR DESCRIPTION
**Pull request by nickvergessen**
## Summary

Missing | Configured
---|---
grafik | grafik

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/settings/lib/SetupChecks/MaintenanceWindowStart.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+	public function getName(): string {
+		return $this->l10n->t('Maintenance window start');
+	}
+
+	public function run(): SetupResult {
+		$configValue = $this->config->getSystemValue('maintenance_window_start', null);
+		if ($configValue === null) {
+			return SetupResult::warning(
+				$this->l10n->t('Server has no maintenance window start time configured. This means resource intensive daily background jobs will also be executed during your main usage time. We recommended to set it to a time of low usage, so users are less impacted by the load caused from these heavy tasks.'),
```
Comment (marcelklehr) [inline apps/settings/lib/SetupChecks/MaintenanceWindowStart.php:55]:
```suggestion
				$this->l10n->t('Server has no maintenance window start time configured. This means resource intensive daily background jobs will also be executed during your main usage time. We recommend to set it to a time of low usage, so users are less impacted by the load caused from these heavy tasks.'),
```

**Review by marcelklehr — APPROVED** [review]

**Review by come-nc — COMMENTED** [review]
nitpick and I agree with @marcelklehr remark, but otherwhise good.
But you will conflict with https://github.com/nextcloud/server/pull/41547 https://github.com/nextcloud/server/pull/42176 https://github.com/nextcloud/server/pull/42177 , wait in line :-P

File: apps/settings/lib/SetupChecks/MaintenanceWindowStart.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		}
+
+		$startValue = (int) $configValue;
+		$endValue = ($startValue + 6) % 24;
+		return SetupResult::success(
+			str_replace(
+				['{start}', '{end}'],
+				[$startValue, $endValue],
+				$this->l10n->t('Maintenance window to execute heavy background jobs is between {start}:00 UTC and {end}:00 UTC')
+			)
```
[the remark this answers is not in the document]
Comment (nickvergessen) [reply apps/settings/lib/SetupChecks/MaintenanceWindowStart.php:67]: Experience shows that it's non-straight-forward for translators and caused issues every now and then.
They all prefer the "JS way" of parameter handling.
I'm always so close to implementing it directly, just lacking the time.

**Comment by nickvergessen** [comment]
> But you will conflict with https://github.com/nextcloud/server/pull/41547 https://github.com/nextcloud/server/pull/42176 https://github.com/nextcloud/server/pull/42177 , wait in line :-P

Only on the autoloader, because I'm not changing JS as it is new, right?
So we can merge this?

**Review by come-nc — APPROVED** [review]
Hum the autoloader and the registration I think, we’ll see.

**Pull request merged**